services/wallet/router/router_v2.go

```diff
@@ -855,6 +893,18 @@ func (r *Router) resolveCandidates(ctx context.Context, input *RouteInputParams,
 						continue
 					}
 
+					// if just a single from and to chain is selected, we can skip the bridge as potential path
+					if input.SendType == Transfer &&
+						(pProcessor.Name() == pathprocessor.ProcessorBridgeHopName ||
+							pProcessor.Name() == pathprocessor.ProcessorBridgeCelerName) &&
+						len(selectedFromChains) == 1 &&
+						len(selectedToChains) == 1 &&
+						arraysWithSameElements(selectedFromChains, selectedToChains, func(a, b *params.Network) bool {
```

Comment:
lots of code for comparing arrays that contain a single item :D

resolveCandidates is quite a chunky function already, this check could be split into testable/maintainable functions `isSimpleTransfer` and `isProcessorBridge`

Reply:
It has to be a lot. :)

Updated, now it's the same amount of code, just moved to more functions, and that condition looks nicer to read.